### PR TITLE
Add EnsureComponentWarn methods

### PR DESCRIPTION
**Pull request by DrSmugleaf**
Ensures a component, if it doesn't exist it logs a warning. A lot of components in content were repeating this pattern manually in different places.

### CODE REVIEW

File: Robust.Shared/GameObjects/ComponentExt.cs [hunk 1]
```diff
@@ -41,5 +42,65 @@ public static T EnsureComponent<T>(this IEntity entity) where T : Component, new
...
+            if (entity.TryGetComponent(out component!))
+            {
+                return true;
+            }
+
+            warning ??= $"Entity {entity} " +
+                        (entity.Prototype == null
+                            ? ""
+                            : $"with prototype id {entity.Prototype.ID} ")
+                        + $"at {entity.Transform.MapPosition} did not have a {nameof(T)}";
```
Comment (PJB3005) [inline Robust.Shared/GameObjects/ComponentExt.cs]: 1. Entities already have their prototype ID in their `ToString()` so you can at least cut that part out.
2. `nameof(T)` just returns `"T"`. You need `typeof(T)` here.

File: Robust.Shared/GameObjects/ComponentExt.cs [hunk 2]
```diff
@@ -41,5 +42,65 @@ public static T EnsureComponent<T>(this IEntity entity) where T : Component, new
...
+            if (entity.TryGetComponent(out T? component))
+            {
+                return component;
+            }
+
+            warning ??= $"Entity {entity} " +
+                        (entity.Prototype == null
+                            ? ""
+                            : $"with prototype id {entity.Prototype.ID} ")
+                        + $"at {entity.Transform.MapPosition} did not have a {nameof(T)}";
```
Comment (PJB3005) [inline Robust.Shared/GameObjects/ComponentExt.cs]: Same problems apply here.

**Review by PJB3005 — CHANGES_REQUESTED** [review]

File: Robust.Shared/GameObjects/ComponentExt.cs (hunk 1 above)
Comment (DrSmugleaf) [reply Robust.Shared/GameObjects/ComponentExt.cs]: Fixed.

File: Robust.Shared/GameObjects/ComponentExt.cs (hunk 2 above)
Comment (DrSmugleaf) [reply Robust.Shared/GameObjects/ComponentExt.cs]: Fixed.

**Comment by DrSmugleaf** [comment]
Reviews addressed.

**Review by PJB3005 — APPROVED** [review]

**Pull request merged**